kedro/config/omegaconf_config.py

```diff
@@ -0,0 +1,262 @@
+            f"config_patterns={self.config_patterns})"
+        )
+
+    @property
+    def conf_paths(self):
+        """Property method to return deduplicated configuration paths."""
+        return _remove_duplicates(self._build_conf_paths())
+
+    def get(self, *patterns: str) -> Dict[str, Any]:  # type: ignore
+        return _get_config_from_patterns_omegaconf(
+            conf_paths=self.conf_paths, patterns=list(patterns)
+        )
+
+    def _build_conf_paths(self) -> Iterable[str]:
+        run_env = self.env or self.default_run_env
+        return [
+            str(Path(self.conf_source) / self.base_env),
+            str(Path(self.conf_source) / run_env),
+        ]
+
+
+def _get_config_from_patterns_omegaconf(
+    conf_paths: Iterable[str],
+    patterns: Iterable[str] = None,
+) -> Dict[str, Any]:
+    """Recursively scan for configuration files, load and merge them using OmegaConf, and
+    return them in the form of a config dictionary.
+
+    Args:
+        conf_paths: List of configuration paths to directories
+        patterns: Glob patterns to match. Files, which names match
+            any of the specified patterns, will be processed.
+
+    Raises:
+        ValueError: If 2 or more configuration files inside the same
+            config path (or its subdirectories) contain the same
+            top-level key.
+        MissingConfigException: If no configuration files exist within
+            a specified config path.
+        ParserError: If configuration is poorly formatted and
+            cannot be loaded.
+
+    Returns:
+        Dict[str, Any]:  A Python dictionary with the combined
+            configuration from all configuration files.
+    """
+
+    if not patterns:
+        raise ValueError(
+            "'patterns' must contain at least one glob "
+            "pattern to match config filenames against."
+        )
+
+    # In the first iteration of the OmegaConfLoader we'll keep the resolver turned-off.
+    # It's easier to introduce them step by step, but removing them would be a breaking change.
+    _clear_omegaconf_resolvers()
+
+    aggregate_config = []
+    processed_files = set()  # type: Set[Path]
+
+    for conf_path in conf_paths:
+        if not Path(conf_path).is_dir():
+            raise ValueError(
+                f"Given configuration path either does not exist "
+                f"or is not a valid directory: {conf_path}"
+            )
+
+        config_filepaths = _lookup_config_filepaths(
+            Path(conf_path), patterns, processed_files, _config_logger, True
+        )
+
+        new_conf = _load_configs_omegaconf(config_filepaths=config_filepaths)
+
+        aggregate_config.append(new_conf)
+        processed_files |= set(config_filepaths)
+
+    if not processed_files:
+        raise MissingConfigException(
+            f"No files of YAML or JSON format found in {conf_paths} matching the glob "
+            f"pattern(s): {patterns}"
+        )
```

Comment:
Almost, but we use `processed_files` to check if stuff has been processed before so that we don't load it again and then we emit a log message: https://github.com/kedro-org/kedro/blob/main/kedro/config/common.py#L182-L187